### PR TITLE
refactor: use object for GraphQLError constructor

### PR DESCRIPTION
**Pull request by n1ru4l**
Follow up for https://github.com/graphql/graphql-js/pull/3454

Implements the suggestion of https://github.com/graphql/graphql-js/pull/3454#issuecomment-1017257345

> **IvanGoncharov** commented 6 days ago
> If you have time can you please, switch all the existing code in this library to use the new constructor signature (including the tests, just leave one test checking that GraphQLError can accept positional arguments).

### CODE REVIEW

File: src/error/GraphQLError.ts [hunk 1]
```diff
@@ -121,8 +122,10 @@ export class GraphQLError extends Error {
     positions?: Maybe<ReadonlyArray<number>>,
     path?: Maybe<ReadonlyArray<string | number>>,
     originalError?: Maybe<Error & { readonly extensions?: unknown }>,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
     extensions?: Maybe<GraphQLErrorExtensions>,
   );
+  // eslint-disable-next-line @typescript-eslint/unified-signatures
   constructor(message: string, args?: GraphQLErrorArgs);
   constructor(message: string, ...rawArgs: BackwardsCompatibleArgs) {
```
Comment (n1ru4l) [inline src/error/GraphQLError.ts:130]: had to do this so the following is not highlighted as deprecated.

```ts
new GraphQLError("foo")
```

new ~~GraphQLError~~("foo")

seems like `@typescript-eslint/unified-signatures` does not work well with `@deprecated`

**Review by dotansimha — APPROVED** [review]

**Review by saihaj — APPROVED** [review]

File: src/validation/rules/custom/NoSchemaIntrospectionCustomRule.ts [hunk 2]
```diff
@@ -28,7 +28,9 @@ export function NoSchemaIntrospectionCustomRule(
         context.reportError(
           new GraphQLError(
             `GraphQL introspection has been disabled, but the requested query contained the field "${node.name.value}".`,
-            node,
+            {
+              nodes: node,
+            },
```
Comment (IvanGoncharov) [inline src/validation/rules/custom/NoSchemaIntrospectionCustomRule.ts:33]: Please format as single lines.
Same as you did here: https://github.com/graphql/graphql-js/pull/3465/files#diff-1b1ca2cbd9180fc2e2431a40a79598a9e92880d044ee3b2146bf05caf52b6ad6R33
Please check all changes.

Comment (n1ru4l) [reply src/validation/rules/custom/NoSchemaIntrospectionCustomRule.ts]: done.

File: src/error/GraphQLError.ts (hunk 1 above)
Comment (glasser) [reply src/error/GraphQLError.ts:130]: I propose to solve this issue in a different way in #3514

Comment (n1ru4l) [reply src/error/GraphQLError.ts]: rebased on top of your change.

**Pull request merged**